backends/tc/backend.cpp

```diff
@@ -0,0 +1,623 @@
+                        t->name.originalName, action->getName().originalName);
+                break;
+            }
+        }
+        if (::errorCount() > 0) {
+            return;
+        }
+        if ((defaultHit > 0) && (defaultHitConst > 0)) {
+            ::error(ErrorType::ERR_INVALID,
+                    "Table '%1%' cannot have both '@default_hit' action "
+                    "and '@default_hit_const' action",
+                    t->name.originalName);
+            return;
+        } else if (defaultHit > 1) {
+            ::error(ErrorType::ERR_INVALID, "Table '%1%' can have only one '@default_hit' action",
+                    t->name.originalName);
+            return;
+        } else if (defaultHitConst > 1) {
+            ::error(ErrorType::ERR_INVALID,
+                    "Table '%1%' can have only one '@default_hit_const' action",
+                    t->name.originalName);
+            return;
+        }
+        if (defaultActionName != nullptr && defaultActionName != "NoAction") {
+            for (auto tcAction : tcPipeline->actionDefs) {
+                if (defaultActionName == tcAction->actionName) {
+                    tabledef->setDefaultHitAction(tcAction);
+                    if (defaultHitConst == 1) {
+                        tabledef->setDefaultHitConst(true);
+                    }
+                }
+            }
+        }
+    }
+}
+
+void ConvertToBackendIR::postorder(const IR::P4Table *t) {
+    if (t != nullptr) {
+        tableCount++;
+        unsigned int tId = tableCount;
+        auto tName = t->name.originalName;
+        tableIDList.emplace(tId, tName);
+        auto ctrl = findContext<IR::P4Control>();
+        auto cName = ctrl->name.originalName;
+        IR::TCTable *tableDefinition = new IR::TCTable(tId, tName, cName, pipelineName);
+        auto tEntriesCount = DEFAULT_TABLE_ENTRIES;
+        auto sizeProperty = t->getSizeProperty();
+        if (sizeProperty && sizeProperty->fitsUint64()) {
+            tEntriesCount = sizeProperty->asUint64();
+        }
+        tableDefinition->setTableEntriesCount(tEntriesCount);
+        unsigned int keySize = 0;
+        unsigned int keyCount = 0;
+        auto key = t->getKey();
+        if (key != nullptr && key->keyElements.size()) {
+            for (auto k : key->keyElements) {
+                auto keyExp = k->expression;
+                auto keyExpType = typeMap->getType(keyExp);
+                auto widthBits = keyExpType->width_bits();
+                keySize += widthBits;
+                keyCount++;
+            }
+        }
+        tableDefinition->setKeySize(keySize);
+        tableKeysizeList.emplace(tId, keySize);
+        auto annoList = t->getAnnotations()->annotations;
+        for (auto anno : annoList) {
+            if (anno->name != "nummask") continue;
+            auto expressionList = anno->expr;
+            for (auto expr : expressionList) {
```

Comment:
can the list have more than 1 value?

Reply:
List can have only one value .updated the code to handle single value